### PR TITLE
Add option to pass Sierra programs to cairo1-run

### PR DESCRIPTION
**Pull request by Vid201**
# TITLE

## Description

This PR adds support to pass Sierra programs directly to cairo1-run. Useful in cases when you already have Sierra representations.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

**Comment by Oppen** [comment]
I see the linter fails due to an unnecessary binding. The comment I left before should fix it (it's the same place). Also, we'll need an updated changelog. The Hyperfine workflow failure is expected, it can't post the comment due to credential issues for PRs from external contributors.

Please fix the changelog and the lint and I think it's good to go.

File: cairo1-run/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Now that you have the dependencies necessary to run the tests, you can run:
 make test
 ```
 
-To execute a cairo 1 program
+To execute a cairo 1 program (you can also execute Sierra program)
```
Comment (Oppen) [inline cairo1-run/README.md:19]: The program is still Cairo 1, right? Maybe rephrase that to "either as casm or Sierra"?

**Review by Oppen — APPROVED** [review]

File: cairo1-run/README.md (hunk 1 above)
Comment (Vid201) [reply cairo1-run/README.md:19]: Yes, but the command accepts now either Cairo source file or Sierra.

**Review by Oppen — DISMISSED** [review]

**Review by fmoletta — APPROVED** [review]

**Review by Oppen — APPROVED** [review]

File: cairo1-run/src/main.rs [hunk 2]
```diff
@@ -221,14 +221,21 @@ fn run(args: impl Iterator<Item = String>) -> Result<Option<String>, Error> {
         append_return_values: args.append_return_values,
     };
 
-    let compiler_config = CompilerConfig {
-        replace_ids: true,
-        ..CompilerConfig::default()
+    // Try to parse the file as a sierra program
+    let file = std::fs::read_to_string(&args.filename)?;
+    let sierra_program = match cairo_lang_sierra::ProgramParser::new().parse(&file) {
```
Comment (fmoletta) [inline cairo1-run/src/main.rs:226]:
```suggestion
    let sierra_program = match serde_json::from_str(&file) {
```

**Comment by Oppen** [comment]
#1719 got merged instead. @fmoletta, don't forget to add a coauthored-by or some other form of credit the next time 🙈

Reference for the PR on its own doesn't give credit to the original author.
